src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.RegularCompilationTracker.cs

```diff
@@ -792,8 +792,7 @@ public ICompilationTracker WithDoNotCreateCreationPolicy(CancellationToken cance
                     var alreadyParsedTrees = alreadyParsedTreesBuilder.ToImmutableAndClear();
                     var lazyCompilationWithoutGeneratedDocuments = new Lazy<Compilation>(() => this.CreateEmptyCompilation().AddSyntaxTrees(alreadyParsedTrees));
 
-                    // Safe cast to appease NRT system.
-                    var lazyCompilationWithGeneratedDocuments = (Lazy<Compilation?>)lazyCompilationWithoutGeneratedDocuments!;
+                    var lazyCompilationWithGeneratedDocuments = new CancellableLazy<Compilation?>(cancellationToken => lazyCompilationWithoutGeneratedDocuments.Value);
```

Comment:
On mobile, so likely missing something. But why do we need the lazy above this cancelable lazy?

Reply:
These two lazys are arguments 2 & 4 to the InProgressState.ctor call below

Reply:
Sounds good!  I imagine there might be ways to clean this up, but def out of scope for this pr.  Thanks for figuring this out!!   We might want to consider backporting.

Reply:
Sounds good!  I imagine there might be ways to clean this up, but def out of scope for this pr.  Thanks for figuring this out!!   We might want to consider backporting.